kentikapi/client.go

```diff
@@ -162,6 +151,44 @@ func NewClient(c Config) (*Client, error) {
 	}, nil
 }
 
+func (c *Config) FillDefaults() {
+	if c.APIURL == "" {
+		c.APIURL = APIURLUS
+	}
+
+	if c.CloudExportAPIURL == "" {
+		c.CloudExportAPIURL = cloudExportAPIURL
+	}
+
+	if c.SyntheticsAPIURL == "" {
+		c.SyntheticsAPIURL = syntheticsAPIURL
+	}
+
+	if c.CloudExportGRPCHostPort == "" {
+		c.CloudExportGRPCHostPort = cloudExportGRPCHostPort
+	}
+
+	if c.SyntheticsGRPCHostPort == "" {
+		c.SyntheticsGRPCHostPort = syntheticsGRPCHostPort
+	}
+
+	if c.Timeout == nil {
+		c.Timeout = durationPtr(defaultTimeout)
+	}
+
+	if c.RetryCfg.MaxAttempts == nil {
```

Comment:
I think that it could be extracted to c.RetryConfig.FillDefaults() that would be called here. That way we can also use that method in httputil.

Reply:
But they work a bit differently, because httputil.go fills the fields RetryMax, RetryWaitMin and RetryWaitMax of the retryablehttp client and client.go fills the fields in the configuration.
I've made some adjustments let me know after the next commit if it's a good approach.

Reply:
Your adjustment is almost what I wanted. I left one small comment in comment above.

Reply:
Done